### ISSUE TITLE
[BUG] `quotes` rule incorrectly reports error for attribute with template expression

### ISSUE THREAD

**Issue by yeonjuan** [yeonjuan/html-eslint#333]
**Describe the bug**
The rule should not enforce quoting style for dynamic expressions like this

```html
const bar = "bar";
html`<div foo=${bar}></div>`;
```

**To Reproduce**

https://html-eslint.org/playground#eyJjb2RlIjoiY29uc3QgYmFyID0gXCJiYXJcIjtcbmh0bWxgPGRpdiBmb289JHtiYXJ9PjwvZGl2PmA7IiwiY29uZmlnIjp7InJ1bGVzIjp7IkBodG1sLWVzbGludC9xdW90ZXMiOiJlcnJvciJ9fSwibGFuZ3VhZ2UiOiJqYXZhc2NyaXB0In0=

**Expected behavior**

No lint error for the above case.

**Additional context**
